hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -200,6 +203,19 @@
       .defaultValue("true")
       .withDocumentation("Similar to " + BLOOM_INDEX_UPDATE_PARTITION_PATH_ENABLE + ", but for simple index.");
 
+  // ***** Bucket Index Configs *****
+  public static final ConfigProperty<Integer> BUCKET_INDEX_BUCKET_NUM = ConfigProperty
+      .key("hoodie.index.bucket.num")
+      .defaultValue(-1)
+      .withDocumentation("Only applies if index type is BUCKET_INDEX. Determine the bucket num of the hudi table, "
+          + "and each partition is divided to N buckets.");
+
+  public static final ConfigProperty<String> INDEX_KEY_FILED_NAME = ConfigProperty
+      .key("hoodie.index.key.field")
+      .noDefaultValue()
+      .withAlternatives("hoodie.datasource.write.indexkey.field")
+      .withDocumentation("Index key.It is used to index the record and find its file group");
```

Comment:
this is the field we hash by? if so, call this `BUCKET_INDEX_HASH_FIELD` ?  and also fix the property name?

Reply:
done